### PR TITLE
pause: Rewrite pause binary in go.

### LINKED ISSUE #339: Rewrite pause binary in go.
Requiring the pause binary be written in C means we need to build it statically. However, this is problematic on RHEL 7, so let's rewrite it in go for now.

See https://github.com/clearcontainers/packaging/issues/38.

### PR DESCRIPTION
**Pull request by jodh-intel**
Requiring a statically-linked C program causes problems on some
platforms (such as RHEL).

Fixes #339.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: pause/pause.go [hunk 1]
```diff
@@ -14,22 +13,37 @@
...
+)
+
+func sigHandler(sig os.Signal) {
+	var ret int
 
-static void sigdown(int signo) {
-	psignal(signo, "shutting down, got signal");
-	exit(0);
+	switch sig {
+	case os.Interrupt:
```
Comment (sboeuf) [inline pause/pause.go]: You could maybe try to be consistent with the following statement by using `syscall.SIGINT`.

File: pause/pause.go [hunk 2]
```diff
@@ -14,22 +13,37 @@
...
-	if (signal(SIGTERM, sigdown) == SIG_ERR)
-		return 2;
-	for (;;) pause();
-	fprintf(stderr, "error: infinite loop terminated\n");
-	return 42;
+func main() {
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch)
+
+	for {
```
Comment (sboeuf) [inline pause/pause.go]: I think you could avoid to buffer the channel `ch` and that way, you don't need the loop `for`. The call to switch in the function signal handler should block until it gets a signal.

File: pause/pause.go (hunk 1 above)
Comment (jodh-intel) [inline pause/pause.go]: Yep - fixing...

File: pause/pause.go (hunk 2 above)
Comment (sboeuf) [inline pause/pause.go]: @jodh-intel what do you think about this one ?

File: pause/pause.go (hunk 1 above)
Comment (sboeuf) [inline pause/pause.go]: Thanks!

**Comment by jodh-intel** [comment]
This code matches the implementation on https://github.com/clearcontainers/runtime/pull/434 (the runtime has its own copy of the pause binary sources).

**Pull request merged**